lib/src/apigateway/openapi-gateway-lambda.ts

```diff
@@ -133,8 +134,9 @@ const addLogGroupForTracing = (
   }
 
   return {
-    deployOptions: props.deployOptions ?? {
-      metricsEnabled: true,
+    deployOptions: {
+      stageName: props.stage,
+      ...(props.deployOptions ? props.deployOptions : { metricsEnabled: true }),
     },
```

Comment:
I don't think the function `addLogGroupForTracing` should be doing this stage setting. Maybe we need a different function for configuring `deployOptions` or a `addStageNameToDeployedStage`.

Reply:
For me it is a bit confusing as well... it also is not possible to define the `deployOptions` when `accessLogEnable` is defined, which seems unrelated to me.